language/move-lang/src/lib.rs

```diff
@@ -716,39 +703,51 @@ fn parse_program(
     }
 
     let res = if errors.is_empty() {
-        Ok((
-            parser::ast::Program {
-                source_definitions,
-                lib_definitions,
-            },
-            source_comments,
-        ))
+        let pprog = parser::ast::Program {
+            source_definitions,
+            lib_definitions,
+        };
+        let pprog = if sources_shadow_deps {
+            shadow_lib_module_definitions(pprog)
+        } else {
+            pprog
+        };
```

Comment:
This is... bad, but nothing should be done about it now. 

My previous work was a hack and now this hack is making me feel a lot worse as time goes on. 

@tzakian, you are likely going to have to insert some sort of similar filtering logic for testing. 

(cc @bob-wilson per our parse tree discussion)

- Eventually, there will likely be another compiler pass between parsing and expansion. 
- Parsing will not generate an AST and will instead make a parse tree
  - This will facilitate things like an auto-formatter and is also nice for other syntactic tooling
  - It will also let us actually parse comments and cleanup the doc comment hackery (its cool hackery but definitely brittle)
- This ParseTree -> AST conversion will be a really good spot to do things like lib shadowing and test filtering
- When that happens this logic should be in that pass maybe and configured with CompilationEnv
- @tzakian, for your test filtering, feel free to shove that logic here, or maybe make a new skeleton pass that does the filtering, and will also do the ParseTree -> AST conversion later. 

TL;DR This is fine for this PR